### PR TITLE
fix(compiler): Fix compilation of closure scope mutable destructuring

### PR DESCRIPTION
**Pull request by ospencer**

### CODE REVIEW

**Review by marcusroberts — APPROVED** [review]

File: compiler/test/suites/pattern_matching.re [hunk 1]
```diff
@@ -357,6 +357,34 @@ describe("pattern matching", ({test, testSkip}) => {
...
+
+        print(b)
+
+        () => b
+      }
+
+      run()
+    |},
+    "2\n",
+  );
```
Comment (phated) [inline compiler/test/suites/pattern_matching.re:387]: These tests have me wondering about this:
```
let run = () => {
        let mut a = 1
        let mut b = 2
        let mut (x, y) = (a, b)
        b = 7
        print(y)
}
run()
```
What happens?

Comment (ospencer) [reply compiler/test/suites/pattern_matching.re:387]: You get `2`!

Comment (phated) [reply compiler/test/suites/pattern_matching.re:387]: Can you add that a test?

Comment (ospencer) [reply compiler/test/suites/pattern_matching.re:387]: Done!

**Review by phated — APPROVED** [review]
Excellent!

**Pull request merged**